packages/graphql/lib/src/scheduler/scheduler.dart

```diff
@@ -57,11 +57,12 @@ class QueryScheduler {
     }
 
     // fetch each query on the interval
-    intervalQueries[interval]!.forEach(queryManager!.refetchQuery);
+    intervalQueries[interval]!
+        .forEach((e) => queryManager!.refetchQuery<TParsed>(e));
```

Comment:
I don't think this will work 🤔

The `intervalQueries` may contain queries of various `TParsed`. When you're refetching, you're basically casting the stored watch-options to `TParsed`, which then might fail.

Instead, we can consider storing closures in the `intervalQueries` which preserves the right generic.

Reply:
may be your are right
will try to implement that